### PR TITLE
made render_images more robust to channel/cmap choices

### PR DESCRIPTION
**Pull request by timtreis**

### CODE REVIEW

File: src/spatialdata_plot/pl/basic.py [hunk 1]
```diff
@@ -319,6 +321,8 @@ def render_images(
             Color to be used for NAs values, if present.
         alpha
             Alpha value for the shapes.
+        quantile_norm
```
Comment (LucaMarconato) [inline src/spatialdata_plot/pl/basic.py:324]: which quantile?

Comment (LucaMarconato) [reply src/spatialdata_plot/pl/basic.py:324]: Found in the private function, I would maybe expose the quantiles to the user via this function

**Review by LucaMarconato — APPROVED** [review]

File: src/spatialdata_plot/pl/basic.py (hunk 1 above)
Comment (timtreis) [reply src/spatialdata_plot/pl/basic.py:324]: It's based on https://github.com/CSBDeep/CSBDeep/blob/deb54947a734b959681ead6bde24c6143805663f/csbdeep/utils/utils.py#L51, don't know their exact reason behind it though. I could just link this function?

Comment (timtreis) [reply src/spatialdata_plot/pl/basic.py:324]: > Found in the private function, I would maybe expose the quantiles to the user via this function

Good idea!

Comment (LucaMarconato) [reply src/spatialdata_plot/pl/basic.py:324]: I would like the function yes.

**Comment by timtreis** [comment]
Yeah, overhauled the entire `render_images` function because there were a bunch of edge cases for the permutations of the `channel`, `palette` and `cmap` options. @MobiTobi found a few bugs

**Comment by timtreis** [comment]
I'll add tests to lock in this behaviour later and will then merge 👌

**Comment by LucaMarconato** [comment]
Perfect thanks. One last question, how do you use images with tests? Do you assert that the plot needs to look like the images you put in the repo? If so, great idea, I need to do it as well when I transform data!

**Comment by timtreis** [comment]
Yeah exactly, it's taken from Squidpy where @giovp uses that a lot. Super nice for end-to-end tests in plotting.

**Comment by MobiTobi** [comment]
Looks good to me.
